rust/src/lib/ip.rs

```diff
@@ -449,10 +449,42 @@ pub(crate) fn is_ipv6_addr(addr: &str) -> bool {
 
 // TODO: Rust offical has std::net::Ipv6Addr::is_unicast_link_local() in
 // experimental.
-fn is_ipv6_unicast_link_local(ip: &str, prefix: u32) -> bool {
+fn is_ipv6_unicast_link_local(ip: &str, prefix: u8) -> bool {
     // The unicast link local address range is fe80::/10.
     is_ipv6_addr(ip)
         && ip.len() >= 3
         && ["fe8", "fe9", "fea", "feb"].contains(&&ip[..3])
         && prefix >= 10
 }
+
+impl std::convert::TryFrom<&str> for InterfaceIpAddr {
+    type Error = NmstateError;
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let mut addr: Vec<&str> = value.split('/').collect();
+        addr.resize(2, "");
+        let ip = addr[0].to_string();
+        let prefix_length = if addr[1].is_empty() {
+            if is_ipv6_addr(&ip) {
+                128
+            } else {
+                32
+            }
+        } else {
+            addr[1].parse::<u8>().map_err(|parse_error| {
+                let e = NmstateError::new(
+                    ErrorKind::InvalidArgument,
+                    format!("Invalid IP address {}: {}", value, parse_error),
+                );
+                log::error!("{}", e);
+                e
+            })?
+        };
+        Ok(Self { ip, prefix_length })
+    }
+}
+
+impl std::convert::Into<String> for &InterfaceIpAddr {
+    fn into(self) -> String {
+        format!("{}/{}", &self.ip, self.prefix_length)
+    }
+}
```

Comment:
I agree with clippy here. Rust documentation recommends always to implement `From<&ip::InterfaceIpAddr>`. IMO, it is more clear for the developers.

Reply:
Fixed. Thanks.